reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java

```diff
@@ -823,7 +836,16 @@ public void onChannelInit(ConnectionObserver observer, Channel channel, @Nullabl
 			boolean needRead = false;
 
 			if (sslProvider != null) {
-				sslProvider.addSslHandler(channel, remoteAddress, SSL_DEBUG);
+				ChannelPipeline pipeline = channel.pipeline();
+				if (redirectHttp2Https) {
```

Comment:
Let's also check that the protocol is not HTTP/2

```suggestion
				if (redirectHttp2Https && ((protocols & h2) != h2)) {
```

Reply:
Done.